src/main/scala/org/apache/spark/sql/hive/llap/LlapContext.scala

```diff
@@ -124,6 +122,44 @@ class LlapContext(sc: SparkContext,
       Seq.empty
     }
   }
+
+  override protected[sql] def executePlan(plan: LogicalPlan): super.QueryExecution = {
+    new this.QueryExecution(plan)
+    super.executePlan(plan)
+  }
+
+  private[sql] class QueryExecution(logicalPlan: LogicalPlan)
+    extends super.QueryExecution(logicalPlan)  {
+
+    /**
+     * Access control for Desc Table Function
+     */
+    val checkDesc = executedPlan match {
```

Comment:
I left here for someone else could use it in the future. if it is not necessary, I can remove it.